pkg/plugin/plugin_v2.go

```diff
@@ -183,12 +184,12 @@ func (p *V2Plugin) Decrypt(ctx context.Context, request *pb.DecryptRequest) (*pb
 
 	result, err := p.svc.Decrypt(ctx, input)
 	if err != nil {
-		select {
-		case p.healthCheck.healthCheckErrc <- err:
-		default:
+		errorType := kmsplugin.ParseError(err).String()
+		if errorType != kmsplugin.KMSErrorTypeCorruption.String() {
+			p.healthCheck.healthCheckErrc <- err
```

Comment:
this can be a blocking call right, just for safety should we keep using the select so that this can be best effort?